gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/pathfinder/AbstractPathFinder.java

```diff
@@ -150,15 +150,13 @@
       destDatasetDescriptorConfig = destDatasetDescriptorConfig.withFallback(getDefaultConfig(this.destNodes.get(0)));
     }
 
-    Class srcdatasetDescriptorClass =
-        Class.forName(srcDatasetDescriptorConfig.getString(DatasetDescriptorConfigKeys.CLASS_KEY));
-    this.srcDatasetDescriptor = (DatasetDescriptor) GobblinConstructorUtils
-        .invokeLongestConstructor(srcdatasetDescriptorClass, srcDatasetDescriptorConfig);
-    Class destDatasetDescriptorClass =
-        Class.forName(destDatasetDescriptorConfig.getString(DatasetDescriptorConfigKeys.CLASS_KEY));
-    this.destDatasetDescriptor = (DatasetDescriptor) GobblinConstructorUtils
-        .invokeLongestConstructor(destDatasetDescriptorClass, destDatasetDescriptorConfig);
+    this.srcDatasetDescriptor = constructDatasetDescriptor(srcDatasetDescriptorConfig);
+    this.destDatasetDescriptor = constructDatasetDescriptor(destDatasetDescriptorConfig);
+  }
 
+  public static DatasetDescriptor constructDatasetDescriptor(Config config) throws ReflectiveOperationException {
```

Comment:
There is already a method getDatasetDescriptor() in StaticFlowTemplate class that does the exact same thing. Can we reuse that method? Also, do you think it might be better to move this method to a utility class?

Reply:
Good point, didn't notice that. I moved it to a new util class.